docs/transactions.md

```diff
@@ -1495,55 +1337,124 @@ Should be sent to trusted nodes only.
 - REST API:
   - GET `/dcl/pki/proposed-revocation-certificates`
 
-### GET_PKI_REVOCATION_DISTRIBUTION_POINT
+### E2E (NOC)
+
+#### ADD_NOC_ROOT
 
 **Status: Implemented**
 
-Gets a revocation distribution point (such as RFC5280 Certificate Revocation List) identified by (VendorID, Label, IssuerSubjectKeyID) unique combination.
-Use [GET_ALL_PKI_REVOCATION_DISTRIBUTION_POINT](#get_all_pki_revocation_distribution_point) to get a list of all revocation distribution points.
+This transaction adds a NOC root certificate owned by the Vendor.
 
+- Who can send
+  - Vendor account
 - Parameters:
-  - vid: `uint16` -  Vendor ID (positive non-zero)
-  - label: `string` -  A label to disambiguate multiple revocation information partitions of a particular issuer.
-  - issuerSubjectKeyID: `string` - Uniquely identifies the PAA or PAI for which this revocation distribution point is provided. Must consist of even number of uppercase hexadecimal characters ([0-9A-F]), with no whitespace and no non-hexadecimal characters., e.g: `5A880E6C3653D07FB08971A3F473790930E62BDB`.
-- CLI command:
-  - `dcld query pki revocation-point --vid=<uint16> --label=<string> --issuer-subject-key-id=<string>`
-- REST API:
-  - GET `/dcl/pki/revocation-points/{issuerSubjectKeyID}/{vid}/{label}`
+  - cert: `string` - The NOC Root Certificate, encoded in X.509v3 PEM format. Can be a PEM string or a file path.
+  - certificate-schema-version: `optional(uint16)` - Certificate's schema version to support backward/forward compatability(default 0)
+  - schemaVersion: `optional(uint16)` - Schema version to support backward/forward compatability(default 0)
+- In State:
+  - `pki/ApprovedCertificates/value/<Subject>/<SubjectKeyID>`
+  - `pki/ApprovedCertificatesBySubject/value/<Subject>`
+  - `pki/ApprovedCertificatesBySubjectKeyID/value/<SubjectKeyID>`
+  - `pki/NocRootCertificates/value/<VID>`
+- CLI Command:
+  - `dcld tx pki add-noc-x509-root-cert --certificate=<string-or-path> --from=<account>`
+- Validation:
+  - the provided certificate must be a root certificate:
+    - `Issuer` == `Subject`
+    - `Authority Key Identifier` == `Subject Key Identifier`
+  - no existing certificate with the same `<Certificate's Issuer>:<Certificate's Serial Number>` combination.
+  - if certificates with the same `<Certificate's Subject>:<Certificate's Subject Key ID>` combination already exist:
+    - the existing certificate must be NOC root certificate
+    - the sender's VID must match the `vid` field of the existing certificates.
+  - the signature (self-signature) and expiration date must be valid.
 
-### GET_PKI_REVOCATION_DISTRIBUTION_POINTS_BY_SUBJECT_KEY_ID
+#### REVOKE_NOC_ROOT
```

Comment:
Please clarify that revoke here means soft-delete (revoke certificates can be queried)

Reply:
Done